acceptance/tests/reports/corrective_change_via_puppet.rb

```diff
@@ -0,0 +1,105 @@
+require 'yaml'
+
+test_name "C98094 - a resource changed via Puppet manifest will not be reported as a corrective change" do
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+  test_file_name = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, test_file_name)
+  tmp_file = {}
+  original_test_data = 'this is my original important data'
+  modified_test_data = 'this is my modified important data'
+  
+  agents.each do |agent|
+    tmp_file[agent.hostname] = agent.tmpfile(tmp_environment)
+  end
+
+  teardown do
+    step 'clean out produced resources' do
+      agents.each do |agent|
+       if tmp_file.has_key?(agent.hostname) && tmp_file[agent.hostname] != ''  
+        on(agent, "rm #{tmp_file[agent.hostname]}", :accept_all_exit_codes => true)
+      end 
+     end
+    end
+  end
+
+  def create_manifest_for_file_resource(file_resource, file_contents, environment_name)
+    manifest = <<MANIFEST
+file { '#{environmentpath}/#{environment_name}/manifests/site.pp':
+  ensure => file,
+  content => '
+\$test_path = \$::fqdn ? #{file_resource}
+file { \$test_path:
+  content => @(UTF8)
+    #{file_contents}
+    | UTF8
+}
+  ',
+}
+MANIFEST
+    apply_manifest_on(master, manifest, :catch_failures => true)
+  end
+
+  step 'create file resource in site.pp' do
+    create_manifest_for_file_resource(tmp_file, original_test_data, tmp_environment)
+  end
+
+  step 'run agent(s) to create the new resource' do
+    with_puppet_running_on(master, {}) do
+      agents.each do |agent|
+        fqdn = agent.hostname
+
+        step 'Run agent once to create new File resource' do
+          on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),:acceptable_exit_codes => 2)
+        end
+
+        step 'Verify the file resource is created' do
+          on(agent, "cat #{tmp_file[fqdn]}").stdout do |file_contents|
+            assert_equal(original_test_data, file_contents, 'file contents did not match expected contents')
+          end
+        end
+
+        step 'Change the manifest for the resource' do
+          create_manifest_for_file_resource(tmp_file, modified_test_data, tmp_environment)
+        end
+
+        step 'Run agent a 2nd time to change the File resource' do
+          on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),:acceptable_exit_codes => 2)
```

Comment:
This currently fails with a '0' exit code because the server on the master has not been HUP'd to load the changes.